src/java/map/BinaryMask.java

```diff
@@ -811,6 +849,62 @@ public BinaryMask getAreasWithinSpecifiedDistanceOfEdges(int distance) {
         return newMask;
     }
 
+    public FloatMask getDistanceFieldFromPoint(int x, int y) {
+        FloatMask distanceField = new FloatMask(getSize(), null, symmetrySettings);
+        for (int a = 0; a < getSize(); a++) {
+            for (int b = 0; b < getSize(); b++) {
+                distanceField.set(a, b, getDistanceBetweenPoints(x, y, a, b));
+            }
+        }
+        return distanceField;
+    }
+
+    public FloatMask getDistanceFieldFromPointInArea(int x, int y) {
+        FloatMask distanceField = new FloatMask(getSize(), null, symmetrySettings);
+        for (int a = 0; a < getSize(); a++) {
+            for (int b = 0; b < getSize(); b++) {
+                if (this.get(a, b)) {
+                    distanceField.set(a, b, getDistanceBetweenPoints(x, y, a, b));
+                }
+            }
+        }
+        return distanceField;
+    }
+
+    public FloatMask getRankedDistanceFieldFromPoint(int x, int y) {
+        FloatMask distanceField = getDistanceFieldFromPoint(x, y);
+        FloatMask rankedDistanceField = new FloatMask(getSize(), null, symmetrySettings);
+        for (int a = 0; a < getSize(); a++) {
+            for (int b = 0; b < getSize(); b++) {
+                rankedDistanceField.set(a, b, distanceField.getRankOfPoint(a, b));
+            }
+        }
+        return rankedDistanceField;
+    }
+
+    public FloatMask getRankedDistanceFieldFromPointInArea(int x, int y) {
+        FloatMask distanceField = getDistanceFieldFromPointInArea(x, y);
+        FloatMask rankedDistanceField = new FloatMask(getSize(), null, symmetrySettings);
+        for (int a = 0; a < getSize(); a++) {
+            for (int b = 0; b < getSize(); b++) {
+                if (this.get(a, b)) {
+                    rankedDistanceField.set(a, b, distanceField.getRankOfPointInArea(a, b, this));
```

Comment:
this will be slow shouldnt need to calculate the rank everytime for everypoint

Reply:
I'm open to better suggestions that don't remove the intended functionality.